crates/cast/bin/tx.rs

```diff
@@ -367,7 +369,51 @@ where
         }
 
         if self.tx.gas.is_none() {
-            self.tx.gas = Some(self.provider.estimate_gas(&self.tx).await?);
+            match self.provider.estimate_gas(&self.tx).await {
```

Comment:
please move to a separate function that also uses let-else, this is too indented to make sense

Reply:
I reorganized code a little bit, please check https://github.com/foundry-rs/foundry/pull/9794/commits/6adb6a8adffd880dc4af82c58e931d1de6a31fe0